src/apps/gearbox/ethereum/gearbox.credit-accounts.contract-position-fetcher.ts

```diff
@@ -0,0 +1,127 @@
+@PositionTemplate()
+export class EthereumGearboxCreditAccountsContractPositionFetcher extends ContractPositionTemplatePositionFetcher<CreditManagerV2> {
+  constructor(
+    @Inject(APP_TOOLKIT) readonly appToolkit: IAppToolkit,
+    @Inject(GearboxContractFactory) private readonly gearboxContractFactory: GearboxContractFactory,
+  ) {
+    super(appToolkit);
+  }
+
+  groupLabel = 'Credit Account';
+
+  getContract(address: string): CreditManagerV2 {
+    return this.gearboxContractFactory.creditManagerV2({ address, network: this.network });
+  }
+
+  async getTokenDefinitions(
+    params: GetTokenDefinitionsParams<CreditManagerV2, DefaultContractPositionDefinition>,
+  ): Promise<UnderlyingTokenDefinition[] | null> {
+    const contract = params.contract;
+    const multicall = params.multicall;
+
+    const underlying = await contract.underlying();
+    const collateralTokensCount = await contract.collateralTokensCount();
+    const collateralTokens = await Promise.all(
+      _.range(collateralTokensCount.toNumber()).map(idx =>
+        multicall
+          .wrap(contract)
+          .collateralTokens(idx)
+          .then(([token, _]) => token),
+      ),
+    );
+
+    return [
+      {
+        address: underlying,
+        metaType: MetaType.BORROWED,
+        network: this.network,
+      },
+      ...collateralTokens.map(token => ({
+        address: token,
+        metaType: MetaType.SUPPLIED,
+        network: this.network,
+      })),
+    ];
+  }
+
+  async getLabel(
+    params: GetDisplayPropsParams<CreditManagerV2, DefaultDataProps, DefaultContractPositionDefinition>,
+  ): Promise<string> {
+    return '';
+  }
+
+  async getDefinitions(params: GetDefinitionsParams): Promise<DefaultContractPositionDefinition[]> {
+    const contractsRegister = this.gearboxContractFactory.contractsRegister({
+      address: '0xa50d4e7d8946a7c90652339cdbd262c375d54d99',
+      network: this.network,
+    });
+
+    const creditManagers = await contractsRegister.getCreditManagers();
+    const creditManagerContracts = creditManagers.map(addr =>
+      this.gearboxContractFactory.creditManagerV2({ address: addr, network }),
+    );
+
+    const multicall = params.multicall;
+    const creditManagerVersions = await Promise.all(
+      creditManagerContracts.map(contract => multicall.wrap(contract).version()),
+    );
+
+    // only v2 credit managers
+    return creditManagers
+      .filter((_, idx) => creditManagerVersions[idx].toNumber() === 2)
+      .map((address, idx) => ({ address, contract: creditManagerContracts[idx] }));
+  }
+
+  async getTokenBalancesPerPosition({
+    address,
+    contractPosition,
+    contract,
+    multicall,
+  }: GetTokenBalancesParams<CreditManagerV2, DefaultDataProps>): Promise<BigNumberish[]> {
+    let creditAccountAddress = '';
+
+    try {
+      creditAccountAddress = await contract.getCreditAccountOrRevert(address);
+    } catch (err) {
+      return [];
+    }
```

Comment:
I suggest you do something this instead of using try/catch

```
const creditAccountAddress = await contract.getCreditAccountOrRevert(address);
if (!creditAccountAddress) return Array(contractPosition.tokens.length).fill(0)
```